nerd/client/nerdapi.go

```diff
@@ -81,38 +80,45 @@ func (nerdapi *NerdAPIClient) url(p string) string {
 	return url.String()
 }
 
-func (nerdapi *NerdAPIClient) handleError(err error, aerr *payload.Error, resp *http.Response) error {
+func (nerdapi *NerdAPIClient) doRequest(s *sling.Sling, successV interface{}) *APIError {
+	e := &payload.Error{}
+	req, err := s.Request()
+	if err != nil {
+		//TODO: should error message include more details like URL, HTTP method and payload (sling is not very verbose in giving detailed error information)?
```

Comment:
Looking at Go and Sling source code, the error here can be caused by using a wrong http method or a invalid urls